libs/langchain/langchain/retrievers/multi_query.py

```diff
@@ -148,7 +149,7 @@ def unique_union(self, documents: List[Document]) -> List[Document]:
         # Create a dictionary with page_content as keys to remove duplicates
         # TODO: Add Document ID property (e.g., UUID)
         unique_documents_dict = {
-            (doc.page_content, tuple(sorted(doc.metadata.items()))): doc
+            (doc.page_content, json.dumps(doc.metadata, sort_keys=True)): doc
```

Comment:
should we make this configurable? could allow users to pass in a `metadata_encoder` which is used here (and that can have a default value of json.dumps)?  am just trying to get ahead of the case where someone wants to use with non-json-serializable metadata. thoughts @hwchase17 ?

Reply:
yup nice idea

Reply:
Great point, I'm working with PGVector & Chroma and right now they don't support documents with non-json-serializable metadata so I didn't mind. 

Nice to get ahead with it.

Reply:
on second thought do we need encoding at all? could we do #10107